src/transform/transform.ts

```diff
@@ -842,6 +842,7 @@ function getRootDiscriminator(obj: ObjectSchema): ObjectSchema {
       if (isObjectSchema(parent) && parent.discriminator) {
         root.language.go!.discriminatorParent = parent.language.go!.discriminatorInterface;
         root = parent;
+        obj = root;
```

Comment:
Is this what's fixing the infinite loop? How does it do it?

Reply:
Yep this is the fix for that.  I've added a comment explaining how this works.

Reply:
Great, thanks! :)